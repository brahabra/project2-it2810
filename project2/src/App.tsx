import Home from "./pages/Home";
import CommitsPage from "./pages/CommitsPage";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate
} from "react-router-dom";
import Navigationbar from "./components/Navigationbar";
import IssuePage from "./pages/IssuePage";
import "bootstrap/dist/css/bootstrap.min.css";
import "./index.css";
import React, { useContext } from "react";
import { ProjectContext } from "./ProjectContext";

function App() {
  const ctx = useContext(ProjectContext);

  return (
    <div className="App">
      <Navigationbar />
      <Router>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/commits" element={(ctx.projectID == "" || ctx.token == "") ? <Navigate to="/" /> :  <CommitsPage/> }/>
          <Route path="/issues" element={(ctx.projectID == "" || ctx.token == "") ? <Navigate to="/" /> :  <IssuePage/> }/>
        </Routes>
      </Router>
    </div>
  );
}

export default App;
