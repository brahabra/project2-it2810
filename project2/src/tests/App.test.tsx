import * as React from 'react'
import App from '../App'

const ReactTestRenderer = require('react-test-renderer');

it('renders correctly', () => {
  const tree = ReactTestRenderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})