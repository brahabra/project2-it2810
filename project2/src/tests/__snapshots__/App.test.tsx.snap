// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand navbar-dark bg-background-color"
    style={
      {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Logo
    </a>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rr-ui-event-key="/commits"
        disabled={false}
        href="/commits"
        onClick={[Function]}
      >
        Commits
      </a>
      <a
        className="nav-link"
        data-rr-ui-event-key="/issues"
        disabled={false}
        href="/issues"
        onClick={[Function]}
      >
        Issues
      </a>
    </div>
  </nav>
  <div
    className="container"
  >
    <br />
    <br />
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="ProjectID"
        type="number"
        value={17381}
      />
    </div>
    <br />
    <br />
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Project Token"
        type="text"
        value="glpat-CRs4epaLyzKdvdpGzE_3"
      />
    </div>
    <br />
    <br />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Submit
    </button>
    <br />
    <br />
  </div>
</div>
`;
